internal/provider/inventory_resource.go

```diff
@@ -0,0 +1,319 @@
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the inventory resource and sets the updated Terraform state on success.
+func (r *InventoryResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data inventoryResourceModel
+	var diags diag.Diagnostics
+
+	// Read Terraform plan data into inventory resource model
+	diags = req.Plan.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Generate request body from inventory data
+	updateRequestBody, diags := data.generateRequestBody()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	requestData := bytes.NewReader(updateRequestBody)
+
+	// Update inventory in AAP
+	updateResponseBody, diags := r.client.Update(data.Url.ValueString(), requestData)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated inventory data into inventory resource model
+	diags = data.parseHTTPResponse(updateResponseBody)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated state
+	diags = resp.State.Set(ctx, data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Delete deletes the inventory resource.
+func (r *InventoryResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data inventoryResourceModel
+	var diags diag.Diagnostics
+
+	// Read current Terraform state data into inventory resource model
+	diags = req.State.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete inventory from AAP
+	_, diags = r.client.Delete(data.Url.ValueString())
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// InventoryResourceModel maps the inventory resource schema to a Go struct.
+type inventoryResourceModel struct {
+	Id           types.Int64          `tfsdk:"id"`
+	Organization types.Int64          `tfsdk:"organization"`
+	Url          types.String         `tfsdk:"url"`
+	Name         types.String         `tfsdk:"name"`
+	Description  types.String         `tfsdk:"description"`
+	Variables    jsontypes.Normalized `tfsdk:"variables"`
+}
+
+// generateRequestBody creates a JSON encoded request body from the inventory resource data.
+func (r *inventoryResourceModel) generateRequestBody() ([]byte, diag.Diagnostics) {
+	// Convert inventory resource data to API data model
+	var organizationId int64
+
+	// Use default organization if not provided
+	if r.Organization.ValueInt64() == 0 {
+		organizationId = 1
+	} else {
+		organizationId = r.Organization.ValueInt64()
+	}
+	inventory := inventoryAPIModel{
+		Organization: organizationId,
+		Name:         r.Name.ValueString(),
+		Description:  r.Description.ValueString(),
+		Variables:    r.Variables.ValueString(),
```

Comment:
these parameters are optional, therefore we should add them to the request only if they are explicitly set in the configuration (based on one test case from your unit tests it seems that it is working like this)

Reply:
Yes, this is because I added `omitempty` to the json struct tags for the `inventoryAPIModel` type. This is Go's built-in way of removing empty fields when JSON-encoding structs.